### PR TITLE
nosqlbench-824

### PR DESCRIPTION
**Pull request by jshook**

### CODE REVIEW

File: virtdata-lib-basics/src/test/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.security.InvalidParameterException;
+import java.util.Arrays;
+import java.util.LongSummaryStatistics;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TriangularStepFunctionTest {
```
Comment (jeffbanks) [inline virtdata-lib-basics/src/test/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunctionTest.java:28]: Nice test coverage here!

**Review by jeffbanks — APPROVED** [review]
Looks good overall, just a few comments and a question.

**Comment by jeffbanks** [comment]
A question on this [graphic](devdocs/sketches/docs-publishing-flow.svg) what event triggers from the staging to the production environment?

File: virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunction.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+public class TriangularStepFunction implements LongUnaryOperator {
+
+    private final Hash hasher = new Hash();
+    private final long median;
+    private final LongUnaryOperator sizer;
+
+    private final long variance;
+
+
+    @Example({"TriangularStepFunction(100,20)","Create a sequence of values where the average and median is 100, but the range of values is between 90 and 120."})
```
[the remark this answers is not in the document]
Comment (jshook) [reply virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunction.java]: fixed in subsequent push

File: virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunction.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            );
+        }
+        this.median = average;
+        this.variance = variance;
+        this.sizer = new HashRange(average-variance,average+variance);
+    }
+
+    TriangularStepFunction(long average) {
+        this(average, average/2);
+//        if (maxOffset>=avgsize) {
```
[the remark this answers is not in the document]
Comment (jshook) [reply virtdata-lib-basics/src/main/java/io/nosqlbench/virtdata/library/basics/shared/from_long/to_long/TriangularStepFunction.java]: removed commented code and also a unit test method which was for illustration only

**Comment by jshook** [comment]
@jeffbanks Just for clarification, I did update the doc with better layout and some details on the triggering events. For now, the checkpoint from docs staging/preview site to main site will be manual, but after that, we can enhance it.

**Pull request merged**